sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala

```diff
@@ -61,6 +62,34 @@ object ScalaJSPlugin extends AutoPlugin {
 
     // Factory methods for JSEnvs
 
+    /** Creates a [[sbt.Def.Initialize Def.Initialize]] for a [[RhinoJSEnv]].
+     *
+     *  Use this to explicitly specify in your build that you would like to run
+     *  with Rhino:
+     *
+     *  {{{
+     *  Seq(Compile, Test).flatMap(c => inConfig(c)(jsEnv := RhinoJSEnv().value))
+     *  }}}
+     *
+     *  The Rhino JS environment will support DOM through `env.js` if and only
+     *  if `scalaJSRequestsDOM.value` evaluates to `true`.
+     *
+     *  Note that the resulting [[sbt.Def.Setting Setting]] must be scoped in a
+     *  project that has the `ScalaJSPlugin` enabled to work properly.
+     *  Therefore, either put the upper line in your project settings (common
+     *  case) or scope it manually, using
+     *  [[sbt.ProjectExtra.inScope[* Project.inScope]].
+     */
+    def RhinoJSEnv(): Def.Initialize[Task[RhinoJSEnv]] = Def.task {
+      /* We take the semantics from the linker, since they depend on the stage.
+       * This way we are sure we agree on the semantics with the linker.
+       */
+      import ScalaJSPluginInternal.{scalaJSLinker, scalaJSRequestsDOM}
+      val semantics = scalaJSLinker.value.semantics
+      val withDOM = scalaJSRequestsDOM.value
+      new RhinoJSEnv(semantics, withDOM)
+    }
+
```

Comment:
It does become deprecated in the second commit, indeed.

The initial idea of this method was to decompose the deprecations into:
1. We deprecated `scalaJSUseRhino`, but there is still an easy, non-deprecated way to enable it.
2. Then, we really deprecate `RhinoJSEnv` itself.

In case people scream about the deprecation of Rhino, we can easily revert the second commit. `scalaJSUseRhino` would still be deprecated.

Does that make sense?